web/src/beta/components/fields/ColorField/hooks.ts

```diff
@@ -0,0 +1,181 @@
+import { useState, useEffect, useCallback, useRef } from "react";
+import tinycolor, { ColorInput } from "tinycolor2";
+
+export type RGBA = {
+  r: number;
+  g: number;
+  b: number;
+  a: number;
+};
+
+export type Params = {
+  value?: string;
+  onChange?: (value: string) => void;
+};
+
+export default ({ value, onChange }: Params) => {
+  const [colorState, setColor] = useState<string>();
+  const [rgba, setRgba] = useState<RGBA>(tinycolor(value).toRgb());
+  const [tempColor, setTempColor] = useState(colorState);
+  const [open, setOpen] = useState(false);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const pickerRef = useRef<HTMLDivElement>(null);
+
+  //Helper functions
```

Comment:
Please move helper functions either in to a utils.ts file inside this component's folder OR in to the global utils directory if it seems appropriate 🙏🏻

Reply:
done